gatekeeper.js

```diff
@@ -41,7 +43,7 @@ class Gatekeeper {
         const pause = time.toZDT(command[0]);
         const triggerTime = pause.minus(delta);
 
-        ctx.timer = actions.ScriptExecution.createTimer(triggerTime, ctx._procCommandGenerator(ctx));
+        ctx.timer = helpers.createTimer(triggerTime, ctx._procCommandGenerator(ctx), null, null, 'gatekeeper');
```

Comment:
I would like to avoid hard coding the name where ever possible. I've had occasion to instantiate more than one gatekeeper in the same rule, and even now have several instances of gatekeeper scattered throughout my rules. If they are all named "gatekeeper" it doesn't really help much. So here too I'd like to add a name to the constructor and pass that to the _procCommandGenerator and then follow the usual "if not defined default to ruleUID and only default to gatekeeper (or something else reasonable) if ruleUID isn't defined too".

Reply:
If the gatekeepers are not instantiated in the same UI rule or the same script file, they have different names, but I see the point of passing the name as a parameter. I‘ll fix that.

Reply:
I updated the PR, the name can be passed to the Gatekeeper instance.
If no name is passed, the ruleUID or the filename + `.gatekeeper` is used.

Reply:
To test, that nothing is broken, you can run the code under the tests folder. Each has their own fairly extensive suite of unit tests. These tests are designed to be copied into the UI as a Script and run that way (watching the logs of course) but I'm sure could be run at load time if they were put into the automation/js folder. 

I have on my todo list incorporating real unit testing at some point.

There won't be tests showing that the new stuff works but, based on the simplicity of the changes I'm not too concerned. But I do want to make sure we haven't broken something.